src/workerd/io/actor-sqlite.c++

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#include "actor-sqlite.h"
+#include <algorithm>
+
+namespace workerd {
+
+kj::OneOf<kj::Maybe<ActorCacheInterface::Value>,
+          kj::Promise<kj::Maybe<ActorCacheInterface::Value>>>
```

Comment:
This might-be-a-promise, might-not-be-a-promise is rather awkward. Couldn't this just return the promise version and have those be immediately resolved if necessary?

Reply:
Ugh, I overlooked that it was a pre-existing API thing. Nevermind. Hopefully maybe this could be refactored someday.

Reply:
Yes. This is done as an optimization so that when `ActorCache` is able to return a result from cache, the caller isn't forced to drop back to the KJ event loop (which otherwise necessitates dropping the isolate lock).

In the case of SQLite, we always produce results synchronously, never returning a Promise.